### DISCUSSION TITLE
Separating image into channels does not export A channel

### DISCUSSION THREAD

**Discussion by Jossie-X** [ImageMagick/ImageMagick#2202] [Help]
Hello,

I would like to separate a VueScan image with 4 channels into the individual channels. It worls nicely for R, G and B, but not for A.
I use `magick VS_Dia_0030.tif -separate -channel A test_IR.tif` to do that.

What do I do wrong?

Hermann-Josef

**Comment by fmw42** [comment]
To get the alpha channel, you can do:

`magick VS_Dia_0030.tif -alpha extract alpha.tif`

**Reply by Jossie-X** [reply]
Thanks a lot. That worked nicely.

But why does my attempt fail? It should work according to the documentation.

Hermann-Josef

**Reply by fmw42** [reply]
Put -channel a before -separate. You have to tell -separate what channel to separate. Processing needs settings like -channel before their corresponding operators like -separate.

`magick image.tif -channel a -separate alpha_channel.tif`

**Comment by snibgo** [comment]
@Jossie-X: why do you select the alpha channel from each separated image? They have only one channel each.

Perhaps you intended to select the alpha channel before separating, not after.

"-separate" works fine for me, with IM v7.0.8-64:
```
f:\web\im>%IMG7%magick toes_holed.png -separate info:

toes_holed.png[0] PNG 267x233 267x233+0+0 8-bit Gray 48148B 0.063u 0:00.017
toes_holed.png[1] PNG 267x233 267x233+0+0 8-bit Gray 48148B 0.078u 0:00.034
toes_holed.png[2] PNG 267x233 267x233+0+0 8-bit Gray 48148B 0.109u 0:00.047
toes_holed.png[3] PNG 267x233 267x233+0+0 8-bit Gray 48148B 0.141u 0:00.064
```

**Comment by GeeMack** [comment]
As mentioned above by fmw42, you can specify that you want just the alpha channel with "-channel A" ahead of the separate operation. If you want to get all four channels as individual images you can do this...

    magick input.tif -channel RGBA -separate output%d.tif

**Reply by fmw42** [reply]
or use

`-channel all`

**Comment by Jossie-X** [comment]
Thanks to all of you! I will change my script accordingly.

Hermann-Josef
